src/core/parser/graph_parser/graph_parser.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "algorithms/gfd/gfd.h"
+#include "algorithms/gfd/graph_descriptor.h"
+
+namespace parser {
+
+std::vector<std::string> Split(std::string str, std::string sep);
+std::vector<Literal> ParseLiterals(std::istream& stream);
+void WriteLiterals(std::ostream& stream, std::vector<Literal> const& literals);

```

Comment:
These functions are needed only for implementation, so it is better not to declare them in the header. 
Don't forget to wrap these function to anonymous namespace in the source file.